### PR TITLE
Write tests for `MantaAPI`

### LINKED ISSUE #56: [Feature] Core tests preparation
### 🚀  Suggested Feature 

before change architectures, tests must be prepared

### 🎉 Expected Result

ready for change code architectures

### LINKED ISSUE #71: [Feature] write test for `MantaApi`
### 🚀  Suggested Feature 

- [x] create `MockBackend`(a mocking backend) for test purpose
- [x] patch `_BaseClient` with mocking client to send all request to our `MockBackend`
- [x] write test codes for `MantaAPI` class

### 🎉 Expected Result

- [x] preparing for future refactoring

### 🔥 Motivation

WE NEED TESTS!

### PR DESCRIPTION
**Pull request by BlueWhaleKo**
### 🚀  Related Issue

To close #56 #71 

### :tada: Details

Tested features
- [x] User
- [x] Team
- [x] Project
- [x] Experiment
- [ ] Artifact

#### Contributor Checklist
 - [x] Did you add documentations for upcoming feature?
 - [x] Did you write necessary tests?
 - [x] Did you verify new and existing tests pass locally with your changes?

#### Reviewer checklist
 - [ ] Did you review the code?
 - [ ] Did you update the CHANGELOG?

### CODE REVIEW

**Comment by BlueWhaleKo** [comment]
Tested features
- [ ] User
- [ ] Team
- [ ] Project
- [ ] Experiment
- [ ] Artifact

**Comment by BlueWhaleKo** [comment]
@joowon-dm-snu 
need your review :)

File: manta_client/api/api.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def create_project(
         """
         description = description or " "  # TODO: API will change it to optional later, delete here
         kwargs = locals()
+        kwargs.pop("self")
```
[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply manta_client/api/api.py]: resolved by https://github.com/coxwave/manta/pull/72/commits/25ef3a43429e09e11cc6207d31c781daac1edc72

File: manta_client/api/api.py [hunk 2]
```diff
@@ -190,6 +191,8 @@ def create_experiment(
         name = name or util.generate_id()
 
         kwargs = locals()
+        kwargs.pop("self")
```
[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply manta_client/api/api.py]: resolved by https://github.com/coxwave/manta/pull/72/commits/25ef3a43429e09e11cc6207d31c781daac1edc72

File: tests/utils/test_mock_backend.py [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    team_id = default_data.team.Id
+    response: Response = backend.get(
+        f"/api/team/{team_id}", headers={"Authorization": f"manta-apikey {default_data.api_key.key}"}
+    )
+
+    assert response.status_code == 200
+    assert response.json == default_data.team.__dict__
+
+
+def test_get_my_team(default_data: DefaultData):
```
[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply tests/utils/test_mock_backend.py]: resolved by https://github.com/coxwave/manta/pull/72/commits/c1458e2344870ac076b18e5d574ad769ce842a1b

File: tests/utils/mock_base_client.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+            raise AttributeError("wrong request_type")
+
+        req_func = getattr(self, request_type)
+
+        res: TestResponse
+        url = self.base_url + endpoint
+        if request_type == "get":
+            res = req_func(url, kwargs=kwargs)
+        else:
+            res = req_func(url, kwargs=kwargs)
```
[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply tests/utils/mock_base_client.py]: resolved by https://github.com/coxwave/manta/pull/72/commits/74f0620bd3f90409192083510a9c9c69bdf1c7bd

File: tests/utils/mock_backend.py [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+        self.name = name or random_string(20)
+        self.status = status or "PENDING"
+        self.memo = memo or random_string(20)
+        self.histories = histories or []
+        self.systems = systems or []
+        self.summary = summary or []
+        self.metadata = metadata or []
+        self.config = config or []
+        self.hyperparameter = hyperparameter or []
+        self.tags = tags or []
```
[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply tests/utils/mock_backend.py:131]: resolved by https://github.com/coxwave/manta/pull/72/commits/6ced512bc80aec63c1bfb26b8033d86621be7c98

[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply tests/utils/mock_backend.py:131]: resolved by https://github.com/coxwave/manta/pull/72/commits/6ced512bc80aec63c1bfb26b8033d86621be7c98

File: tests/test_manta_api.py [hunk 6]
```diff
@@ -13,106 +18,291 @@
...
-    stats = []
-    api.send_experiment_record(stats=stats)
-    logs = []
-    api.send_experiment_record(logs=logs)
+#     hist = []
+#     manta_api.send_experiment_record(hist)
+#     stats = []
+#     manta_api.send_experiment_record(stats=stats)
+#     logs = []
+#     manta_api.send_experiment_record(logs=logs)
```
[the remark this answers is not in the document]
Comment (BlueWhaleKo) [reply tests/test_manta_api.py:346]: resolved by https://github.com/coxwave/manta/pull/72/commits/f025d5369e1b09346a36c5469c6646b204b6fd1d

**Review by joowon-dm-snu — APPROVED** [review]
great ! 🥇

**Comment by joowon-dm-snu** [comment]
@BlueWhaleKo Could you resolve conflicts?

**Pull request merged**